File: leetcode/0049. Group Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap();
        for(String str : strs){
            char[] strArr = str.toCharArray();
            Arrays.sort(strArr);
            String sortedStr = String.valueOf(strArr);
            
            groups.putIfAbsent(sortedStr, new ArrayList<String>());
            groups.get(sortedStr).add(str);
        }
        return new ArrayList<>(groups.values());
    }
}

File: leetcode/0049. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;
        for (const string& str : strs) {
            string strCopy{str};
            sort(strCopy.begin(), strCopy.end());
            groups[strCopy].push_back(str);
        }

        vector<vector<string>> ans;
        for (const auto& [_, group] : groups) {
            ans.push_back(group);
        }
        return ans;
    }
};